test/tape/pool.js

```diff
@@ -30,51 +30,37 @@ test(`pool evicts dead resources when factory.validate rejects`, (t) => {
       return true;
     },
   });
-
-  Bluebird.resolve(Array.from(Array(5)))
-    .map(() => {
-      return pool.acquire().promise.catch((e) => {
-        t.fail('1# Could not get resource from pool');
-      });
-    })
-    .map((con) => {
-      pool.release(con);
-      return con;
-    })
-    .map((con) => {
-      // fake kill connections
-      con.error = 'connection lost';
-      return con;
-    })
-    .map((con) => {
-      return pool
-        .acquire()
-        .promise.then((con) => {
-          t.ok(con.id > 4, 'old dead connections were not reused');
-          return con;
+  const fiveElements = Array.from(Array(5));
+  const allocateConnections = () =>
+    Promise.all(
+      fiveElements.map(() =>
+        pool.acquire().promise.catch((e) => {
+          t.fail('Could not get resource from pool', e);
         })
-        .catch((e) => {
-          t.fail('2# Could not get resource from pool');
-        });
-    })
-    .map((con) => {
-      pool.release(con);
-      return con;
-    })
-    .map((con) => {
-      return pool.acquire().promise.then((con) => {
-        t.ok(con.id > 4 && con.id < 11, 'Released working connection was used');
-        return con;
-      });
-    })
-    .map((con) => {
-      pool.release(con);
-      return con;
-    })
-    .then(() => pool.destroy())
-    .then(() => {
-      t.end();
-    });
+      )
+    );
+
+  const connectionsGroup1 = await allocateConnections();
+  connectionsGroup1.forEach((con) => pool.release(con));
+  connectionsGroup1.forEach((con) => {
+    // fake kill connections
+    con.error = 'connection lost';
+  });
```

Comment:
It's not important, but it looks like these `.forEach(..)` statements could be combined.